### PR TITLE
DEV: Make overrides_email an integration spec

### PR DESCRIPTION
**Pull request by davidtaylorhq**
We're moving the location of the email-overriding logic in https://github.com/discourse/discourse/pull/15378, which makes the old unit test in this plugin fail. This commit makes it an integration test, so that it's more robust against core changes, and will continue to pass before and after the core changes.

### CODE REVIEW

File: spec/integration/overrides_email_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      extra: {
+        raw_info: OmniAuth::AuthHash.new(
+          email_verified: true
+        )
+      },
+      credentials: OmniAuth::AuthHash.new
+    )
+  end
+
+  it 'updated user email if enabled' do
```
Comment (nattsw) [inline spec/integration/overrides_email_spec.rb:33]:
```suggestion
  it 'updates user email if enabled' do
```

Comment (nattsw) [reply spec/integration/overrides_email_spec.rb:33]: Also, is there an opposing test case?

Comment (davidtaylorhq) [reply spec/integration/overrides_email_spec.rb]: > Also, is there an opposing test case?

Right now it's built-in to this test case (I toggle the site setting halfway through). Let me split it up to make it clearer 👀

Comment (nattsw) [reply spec/integration/overrides_email_spec.rb]: ahaaaa

**Review by nattsw — APPROVED** [review]

**Pull request merged**